File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string result = "";
        int carry = 0;
        
        // Start from the end of both strings
        int i = a.size() - 1, j = b.size() - 1;
        
        while (i >= 0 || j >= 0 || carry) {
            int sum = carry;
            
            // Add the current bit of 'a' if exists
            if (i >= 0) {
                sum += a[i] - '0';
                i--;
            }
            
            // Add the current bit of 'b' if exists
            if (j >= 0) {
                sum += b[j] - '0';
                j--;
            }
            
            // Append the current bit to result
            result += to_string(sum % 2);
            
            // Update the carry
            carry = sum / 2;
        }
        
        // Since the result is built in reverse order, reverse it at the end
        reverse(result.begin(), result.end());
        return result;
    }
};

File: 0067-add-binary/0067-add-binary.js
var addBinary = function(a, b) {
    let i = a.length - 1;
    let j = b.length - 1;
    let carry = 0;
    let result = '';
    
    // Process both strings from the end towards the beginning
    while (i >= 0 || j >= 0 || carry) {
        let sum = carry;
        
        // Add the corresponding bit of a if i is valid
        if (i >= 0) {
            sum += a[i] - '0'; // Convert character to integer
            i--;
        }
        
        // Add the corresponding bit of b if j is valid
        if (j >= 0) {
            sum += b[j] - '0'; // Convert character to integer
            j--;
        }
        
        // Add the sum's last bit to the result and update carry
        result = (sum % 2) + result;
        carry = Math.floor(sum / 2);
    }
    
    return result;
};
